src/u192_decimal.rs

```diff
@@ -139,6 +154,12 @@ impl From<u128> for Decimal {
     }
 }
 
+impl From<Decimal> for u128 {
```

Comment:
There's a better way of doing this:

```
impl TryRound<u128> for Decimal {
    fn try_round(&self) -> Result<u128> {
        let rounded_val = Self::half_wad()
            .checked_add(self.0)
            .ok_or(DecimalError::MathOverflow)?
            .checked_div(Self::wad())
            .ok_or(DecimalError::MathOverflow)?;
        Ok(u128::try_from(rounded_val)
            .map_err(|_| DecimalError::MathOverflow)?)
    }

    fn try_ceil(&self) -> Result<u128> {
        let ceil_val = Self::wad()
            .checked_sub(U192::from(1u128))
            .ok_or(DecimalError::MathOverflow)?
            .checked_add(self.0)
            .ok_or(DecimalError::MathOverflow)?
            .checked_div(Self::wad())
            .ok_or(DecimalError::MathOverflow)?;
        Ok(u128::try_from(ceil_val).map_err(|_| DecimalError::MathOverflow)?)
    }

    fn try_floor(&self) -> Result<u128> {
        let ceil_val = self
            .0
            .checked_div(Self::wad())
            .ok_or(DecimalError::MathOverflow)?;
        Ok(u128::try_from(ceil_val).map_err(|_| DecimalError::MathOverflow)?)
    }
}
```

Reply:
Ah yeah, I prefer this way, as well. Will refactor accordingly !

Reply:
I added the requested changes, please resolve the conversation

Reply:
please remove this `impl From<Decimal> for u128` so that we keep only one way of doing the conversion

Reply:
Addressed it, also removed unused code (cargo clippy).